newrelic/resource_newrelic_synthetics_private_location.go

```diff
@@ -0,0 +1,193 @@
+package newrelic
+
+import (
+	"context"
+	"log"
+
+	"github.com/newrelic/newrelic-client-go/pkg/common"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/synthetics"
+)
+
+func resourceNewRelicSyntheticsPrivateLocation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicSyntheticsPrivateLocationCreate,
+		ReadContext:   resourceNewRelicSyntheticsPrivateLocationRead,
+		UpdateContext: resourceNewRelicSyntheticsPrivateLocationUpdate,
+		DeleteContext: resourceNewRelicSyntheticsPrivateLocationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
```

Comment:
This is optional in other resources: https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_one_dashboard.go#L37